projects/Funkwhale/services/funkwhale/module.nix

```diff
@@ -0,0 +1,558 @@
+      }
+      (lib.mkIf cfg.createDatabaseLocally {
+        DATABASE_HOST = "";
+        DATABASE_PORT = 5432;
+        DATABASE_USER = cfg.user;
+        # Peer authentication.
+        DATABASE_PASSWORD = "";
+      })
+      (lib.mkIf cfg.createRedisLocally {
+        CACHE_URL = "unix://${config.services.redis.servers.funkwhale.unixSocket}?db=0";
+        CELERY_BROKER_URL = "redis+socket://${config.services.redis.servers.funkwhale.unixSocket}?virtual_host=0";
+      })
+      (lib.mkIf cfg.configureNginx {
+        REVERSE_PROXY_TYPE = "nginx";
+        NGINX_MAX_BODY_SIZE = lib.mkDefault "100M";
+      })
+    ];
+
+    users.users = lib.mkIf (cfg.user == "funkwhale") {
+      ${cfg.user} = {
+        isSystemUser = true;
+        group = cfg.group;
+      };
+    };
+    users.groups = lib.mkIf (cfg.group == "funkwhale") {
+      ${cfg.group} = { };
+    };
+
+    environment.systemPackages = [
+      (pkgs.writeShellScriptBin "funkwhale-manage" ''
+        if [[ ! "$USER" == "${cfg.user}" ]]; then
+          echo "This has to be run with the \`${cfg.user}\` user. Ex: \`sudo -u ${cfg.user} funkwhale-manage\`"
+          exit 1
+        fi
+
+        # `troi.utils.discover_patches()` will error, because it looks for patches
+        # in the current directory.
+        stat . &> /dev/null || echo -e \
+          "WARNING: funkwhale-manage should be run from a directory the ${cfg.user} user has read access to.\n"
+
+        # Set `cfg.settings` environment variables.
+        ${lib.concatMapAttrsStringSep "\n" (
+          name: value: ''export ${name}=${lib.escapeShellArg value}''
+        ) cfg.settings}
+
+        # Set secret environment variables.
+        set -o allexport
+        if [ -f "/var/lib/funkwhale/config/django_secret_key.env" ]; then
+          source "/var/lib/funkwhale/config/django_secret_key.env"
+        fi
+        ${lib.optionalString (cfg.secretEnvironmentFile != null) ''
+          source ${cfg.secretEnvironmentFile}
+        ''}
+        set +o allexport
+
+        exec ${lib.getExe cfg.package} "$@"
+      '')
+    ];
+
+    systemd.tmpfiles.settings.funkwhale = {
+      # Where auto-generated secrets will be.
+      "/var/lib/funkwhale/config".d = {
+        user = cfg.user;
+        group = cfg.group;
+        mode = "0700";
+      };
+
+      # Niscellaneous API state files.
+      "/var/lib/funkwhale/api".d = {
+        user = cfg.user;
+        group = cfg.group;
+        mode = "0700";
+      };
+
+      # Files served by the API.
+      "/var/lib/funkwhale/data".d = {
+        user = cfg.user;
+        group = cfg.group;
+        mode = "0755";
+      };
+      # Default `MEDIA_ROOT`.
+      "/var/lib/funkwhale/data/media".d = {
+        user = cfg.user;
+        group = cfg.group;
+        mode = "0755";
+      };
+      # Default `MUSIC_DIRECTORY_PATH`.
+      "/var/lib/funkwhale/data/music".d = {
+        user = cfg.user;
+        group = cfg.group;
+        mode = "0755";
+      };
+    };
```

Comment:
Use systemd `StateDirectory` to create `/var/lib/funkwhale` directory with correct ownership and permissions.